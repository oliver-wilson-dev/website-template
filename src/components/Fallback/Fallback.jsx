import { useEffect } from 'react';

const Fallback = ({ setLoading }) => {
  useEffect(() => {
    setLoading(true);

    return () => setLoading(false);
  }, []);

  return null;
};

export default Fallback;
