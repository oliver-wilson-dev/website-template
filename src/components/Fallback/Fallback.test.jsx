import React from 'react';
import { mount } from 'enzyme';
import Fallback from './Fallback';

let mockCleanup;

jest.mock('react', () => ({
  ...jest.requireActual('react'), // use actual for all non-hook parts
  useEffect: (fn) => {
    mockCleanup = fn();
  },
}));

const defaultProps = {
  setLoading: jest.fn()
};

const render = () => mount(<Fallback {...defaultProps} />);

describe('Fallback', () => {
  it('renders correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it("should call setLoading with 'true'", () => {
    render();

    expect(defaultProps.setLoading).toHaveBeenCalledWith(true);
  });
  describe('when unmounting', () => {
    it("should call setLoading with 'false'", () => {
      render();

      mockCleanup();
      expect(defaultProps.setLoading).toHaveBeenCalledWith(false);
    });
  });
});
