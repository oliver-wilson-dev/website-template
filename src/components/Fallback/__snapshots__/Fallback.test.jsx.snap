// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fallback renders correctly 1`] = `
<Fallback
  setLoading={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;
